### ISSUE TITLE
Shelly PRO 3EM values not showing up in "Pages" View and VRM

### ISSUE THREAD

**Issue by stenub** [victronenergy/dbus-shelly#4]
Hello,

i have a Shelly PRO 3EM and VenusOS 2.93 running on a Beaglebone Black. I have managed to install the GEN2 driver and configured the Shelly to send the values via websocket to the Beaglebone. 
The Shelly is showing up in the device list and everything seems fine. The Shelly device is selectable and proper values for every phase are shown:
![grafik](https://user-images.githubusercontent.com/16527924/230415885-0129830b-a551-41ae-a1bc-7814170e2e6d.png)
![grafik](https://user-images.githubusercontent.com/16527924/230416626-713ae45a-b454-4979-b249-745c4e421caf.png)

The device role is grid meter.
The problem is that the pages view is showing up as if there is no grid meter. Also in the VRM portal no values are shown.

![grafik](https://user-images.githubusercontent.com/16527924/230417517-4f480250-53f9-4e24-b4e4-3291dcf4d0f8.png)

I updated the VenusOS to the latest RC but the result is the same... @izak What am i doing wrong? Can somebody give me a little hint?

Many thanks!
Kind Regards!
Stefan

**Comment by izak** [comment]
Check Settings -> System Setup -> Ac Input 1. This is probably "Unknown". The overview won't know that you have a grid connection if that is not correctly set.

But you mention you're on a Beaglebone black. So there is no built-in MK3 interface (for the Multi), and unless you plug one in via USB, that setting won't exist (because it is created by mk2-dbus when it first starts up).

So I don't know exactly how you have it set up, but the overview screen may not work correctly if you don't have an mk3-usb plugged in and the above setting correctly configured.

**Comment by stenub** [comment]
Hello @izak! Many thanks for your response! I checked the AC Input1 Setting and it is set to Grid. The setting is available even without the MK3 interface connected..
![grafik](https://user-images.githubusercontent.com/16527924/232033147-bc4d9593-b197-4ef9-b504-346167782b4e.png)

After that i connected my MK3 Interface, it gets recognized, but unfortunately that changes nothing at the pages view.. 
Still no values there... Any further hints?

**Comment by ajnadox** [comment]
Hi! @izak i think this is the same issue that we see in the venus os 3, AC is showing 0w and not displaying the Shellys value even if it's showing correct in the meny https://community.victronenergy.com/questions/210047/shelly-pro-3em-as-grid-meter-with-venus-os-30-and.html

**Comment by EvilPutz** [comment]
Hi, i also have the same problem. Values are showed but it is not apeearing in the Dashboard as PV Inverter

**Comment by pipelka** [comment]
Hi,
i can confirm that a similar thing happens on a RasPi 4 with VenusOS 3.0 connected to a Multiplus 48/3000 (via MK3).
For me only L1 is shown on the AC Input side.
All phase values are shown in the Shelly Energy Meter menu.

**Comment by amcsSH** [comment]
Hi, to confirm this issue also, please check my comment at https://community.victronenergy.com/questions/210047/shelly-pro-3em-as-grid-meter-with-venus-os-30-and.html?childToView=224780#comment-224780 .

**Comment by moonilightshaker** [comment]
Hi, i have the same issue. 
Venus OS: v3.01

**Comment by Duester64** [comment]
Finally I found a solution. I deleted the websocket connection by erasing the websocket adress in the settings an added a script found here: [https://github.com/funkmaster86/dbus-shelly-pro-3em-smartmeter/tree/main](url)
Now it works.

**Comment by Duester64** [comment]
[https://github.com/funkmaster86/dbus-shelly-pro-3em-smartmeter](url)

**Comment by Duester64** [comment]
https://github.com/funkmaster86/dbus-shelly-pro-3em-smartmeter

**Comment by Patrick38110** [comment]
> Hallo Leute, ich habe hier eine Lösung gefunden. Ich habe den Websocket-Eintrag in den Einstellungen gelöscht und ein unten gezeigtes Skript hinzugefügt. Jetzt wird es korrekt angezeigt.
> 
> https://github.com/funkmaster86/dbus-shelly-pro-3em-smartmeter
> 
> ![Bild](https://user-images.githubusercontent.com/61905746/265283037-6db6b3b8-4805-48d3-a5d4-3634efb47096.png)

Hallo Duester64, kann ich direkt das Skript von Funkmaster nehmen? Welches Skript/datei muss ich ersetzten?

**Comment by Duester64** [comment]
Ja, die Werte am AC Eingang saldieren wie Du im Screenshot siehst, es werden alle Phasen verrechnet Installiert habe ich es mit dem Script aus der readme Datei, copy +paste.  Gruß

**Comment by Patrick38110** [comment]
@Duester64, bei den Aktualwerten in der Übersicht soll es auch funktionieren. Nur die Energiewerte in KW/h sollen eventuell nicht richtig angezeigt werden. Das heißt wenn ich direkt das Skript von Funkmaster86 installiere brauche ich keinen Websocket Eintrag löschen? Falls ja wo muss ich diesen löschen? Grüße

**Comment by glipschitz** [comment]
I am able to install this, but the GX does not seem to want to trigger the ESS using this data as the grid value :(

![image](https://github.com/victronenergy/dbus-shelly/assets/46615132/4554f1c7-3d57-4f53-8c62-124604f9e048)

Is there any other settings which need to be set? 

AC1-In is set to Grid.

The dashboard shows the correct data and is reporting to VRM.

Just cannot get the battery to discharge in ESS mode.

**Comment by amadirnjul** [comment]
> Hello,
> 
> i have a Shelly PRO 3EM and VenusOS 2.93 running on a Beaglebone Black. I have managed to install the GEN2 driver and configured the Shelly to send the values via websocket to the Beaglebone. The Shelly is showing up in the device list and everything seems fine. The Shelly device is selectable and proper values for every phase are shown: ![grafik](https://user-images.githubusercontent.com/16527924/230415885-0129830b-a551-41ae-a1bc-7814170e2e6d.png) ![grafik](https://user-images.githubusercontent.com/16527924/230416626-713ae45a-b454-4979-b249-745c4e421caf.png)
> 
> The device role is grid meter. The problem is that the pages view is showing up as if there is no grid meter. Also in the VRM portal no values are shown.
> 
> ![grafik](https://user-images.githubusercontent.com/16527924/230417517-4f480250-53f9-4e24-b4e4-3291dcf4d0f8.png)
> 
> I updated the VenusOS to the latest RC but the result is the same... @izak What am i doing wrong? Can somebody give me a little hint?
> 
> Many thanks! Kind Regards! Stefan

How did you install gen2 drivers?

**Comment by Duester64** [comment]
Have you set up  the config.ini with the right ip? 

[DEFAULT]
AccessType = OnPremise
SignOfLifeLog = 1
CustomName = Shelly Pro 3EM
DeviceInstance = 40
# Role Could be pvinverter or grid
Role = grid
# Available Postions: 0 = AC, 1 = AC-Out 1, AC-Out 2
Position = 0
# Available values see https://docs.python.org/3/library/logging.html#levels
LogLevel=ERROR

[ONPREMISE]
Host=192.168.178.105
Username=
Password=

**Comment by jjdejong** [comment]
> Have you set up the config.ini with the right ip?
> 
> [DEFAULT] AccessType = OnPremise SignOfLifeLog = 1 CustomName = Shelly Pro 3EM DeviceInstance = 40
> # Role Could be pvinverter or grid
> 
> Role = grid
> # Available Postions: 0 = AC, 1 = AC-Out 1, AC-Out 2
> 
> Position = 0
> # Available values see https://docs.python.org/3/library/logging.html#levels
> 
> LogLevel=ERROR
> 
> [ONPREMISE] Host=192.168.178.105 Username= Password=

That is for another implementation not included by Victron. This thread is about Victron's dbus-shelly that is included in recent Victron devices.
